### ISSUE TITLE
[Feature request] Allow changing the default client used by the static methods

### ISSUE THREAD

**Issue by emmericp** [josueeduardo/rest-client#87]
It would be nice to be able to change the client used by the default static methods.

The following would be sufficient:

```
    public synchronized static void setClient(RestClient client) {
        close();
        this.client = client;
    }
```

**Comment by josueeduardo** [comment]
Hi, thanks for the suggestion.
Is there any reason why you can't use the `RestClient` class ? like
`RestClient client = RestClient.builder().build()`

Assigning it to a static variable should give you the same behaviour.

**Comment by emmericp** [comment]
Yeah, that's actually what I'm doing at the moment. I've just noticed this while porting my code from Unirest; I had expected RestClient do be a drop-in replacement but I couldn't do the proxy config.

**Comment by josueeduardo** [comment]
Yeah, make sense to add it back. I've Added a comment to your PR, happy to merge an release it on 1.6.1 alongside few other changes.
